flake.nix

```diff
@@ -401,25 +402,25 @@
         };
       };
 
-      # Build the Nickel VSCode extension. The extension seems to be required
-      # for the LSP to work.
-      vscodeExtension =
-        let node-package = (pkgs.callPackage ./lsp/client-extension { }).package;
-        in
-        (node-package.override rec {
-          pname = "nls-client";
-          outputs = [ "vsix" "out" ];
-          nativeBuildInputs = with pkgs; [
-            # `vsce` depends on `keytar`, which depends on `pkg-config` and `libsecret`
-            pkg-config
-            libsecret
-          ];
-          postInstall = ''
-            npm run compile
-            mkdir -p $vsix
-            echo y | npx vsce package -o $vsix/${pname}.vsix
-          '';
-        }).vsix;
+      # Build the Nickel VSCode extension
+      vscodeExtension = pkgs.mkYarnPackage {
+        pname = "vscode-nickel";
+        src = pkgs.lib.cleanSource ./lsp/client-extension;
+
+        buildPhase = ''
+          export HOME="$TMPDIR"
```

Comment:
It's trying to create `.yarn` in `HOME`. There's probably a better way to fix this but this works for now.